### ISSUE TITLE
How to use the CrossEntropy loss?

### ISSUE THREAD

**Issue by t-artikov** [NVlabs/tiny-cuda-nn#93]
I'm trying to approximate an image that consists of classes rather than arbitrary colors. CrossEntropy loss should work better than L2 for this task, but doesn't work at all for some reason.

See my modification of the mlp_learning_an_image example, please.
https://github.com/t-artikov/tiny-cuda-nn/commit/815319ef2380624dca5310dfa210356d84624676
I use an image that consists of red (1, 0, 0), green (0, 1, 0) and blue (0, 0, 1) colors, which I believe can be interpreted as one-hot encoded classes. I specify CrossEntropy loss and Sigmoid output activation function (to clamp probability in the 0..1 interval).

After training I get loss == 0 and a completely white image. What am I doing wrong?

**Comment by Tom94** [comment]
Thanks for the update and for permission to adapt your code -- a categorical cross entropy loss would indeed be nice to have in the library, so I'll put it onto the todo list.
